src/react/hooks/useFragment.ts

```diff
@@ -112,9 +113,11 @@ function useFragment_<TData = any, TVars = OperationVariables>(
         result: client["queryManager"].maskFragment({
           fragment,
           fragmentName,
-          data: diff.result,
+          // TODO: Revert to `diff.result` once `useFragment` supports `null` as
+          // valid return value
+          data: diff.result === null ? {} : diff.result,
         }),
-      }),
+      } as Cache.DiffResult<TData>),
```

Comment:
These casts look like some pretty blatant lies to the compiler. Is there any way we get around those?

Reply:
```suggestion
      } as Cache.DiffResult<TData>), // TODO: remove assertion
```

Since we're going to touch this code again in the near future, maybe leave a TODO for now?

Reply:
Yep good call.

Reply:
[d9fd2e5](https://github.com/apollographql/apollo-client/pull/12304/commits/d9fd2e5223851ae5fd17939cbe582393a09487bf)